src/operations/map/IpcChoroplethOperations.tsx

```diff
@@ -0,0 +1,166 @@
+import { Feature, FeatureCollection, GeoJsonProperties, Geometry } from 'geojson';
+import L from 'leaflet';
+import { createRoot } from 'react-dom/client';
+
+import CountryHoverPopover from '@/components/CountryHoverPopover/CountryHoverPopover';
+import container from '@/container';
+import { CountryData } from '@/domain/entities/country/CountryData';
+import { CountryIpcData } from '@/domain/entities/country/CountryIpcData';
+import { CountryMapData, CountryMapDataWrapper } from '@/domain/entities/country/CountryMapData';
+import CountryRepository from '@/domain/repositories/CountryRepository';
+
+export class IpcChoroplethOperations {
+  static ipcGlobalStyle = (feature: Feature<Geometry, GeoJsonProperties> | undefined) => ({
+    color: '#000',
+    weight: 0.5,
+    fillOpacity: 1,
+    fillColor: IpcChoroplethOperations.fillGlobalIpc(feature?.properties?.ipcPopulation),
+  });
+
+  static fillGlobalIpc = (ipcPopulation: number | null) => {
+    if (ipcPopulation === null) return 'none';
+    if (ipcPopulation < 0.1) return '#F6D1C1';
+    if (ipcPopulation < 0.5) return '#FC9B7D';
+    if (ipcPopulation < 1.0) return '#FB7453';
+    if (ipcPopulation < 3.0) return '#F24634';
+    if (ipcPopulation < 5.0) return '#D11F26';
+    if (ipcPopulation < 10) return '#AE151B';
+    return '#710013';
+  };
+
+  static ipcCountryStyle = (feature: Feature<Geometry, GeoJsonProperties> | undefined) => ({
+    color: '#fff',
+    opacity: 0.8,
+    weight: 0.5,
+    fillOpacity: 1,
+    fillColor: IpcChoroplethOperations.fillCountryIpc(feature?.properties?.ipcPhase),
+  });
+
+  static fillCountryIpc = (phase: number) => {
+    if (phase === null) return 'RGBA(58, 66, 68, 0.7)';
+    if (phase === 1) return '#D1FAD1';
+    if (phase === 2) return '#FAE834';
+    if (phase === 3) return '#E88519';
+    if (phase === 4) return '#CD1919';
+    if (phase === 5) return '#731919';
+    if (phase === 6) return '#353F42';
+    return 'RGBA(58, 66, 68, 0.7)';
+  };
+
+  static generateColorMap = (ipcData: CountryIpcData[], mapData: CountryMapDataWrapper) => {
+    const ipcDataById = Object.fromEntries(ipcData.map((data: CountryIpcData) => [data.adm0_code, data]));
+
+    const updatedFeatures = mapData.features.map((feature: CountryMapData) => ({
+      ...feature,
+      properties: {
+        ...feature.properties,
+        ipcData: ipcDataById[feature.properties.adm0_id] || null,
+      },
+    }));
+
+    return { type: 'FeatureCollection', features: updatedFeatures as Feature<Geometry, GeoJsonProperties>[] };
+  };
```

Comment:
would it be possible to filter here so that only those countries are returned where there's IPC data? It looks like a second layer is added to the non-ipc countries as well (the border gets thicker)

Reply:
yeah thats true actually. Nice observation 👏 👏